src/lib.rs

```diff
@@ -251,40 +258,76 @@ impl PerformanceData {
-            let ret = tfd.read();
-            if ret > 1 {
-                error!("Missed {} interval(s)", ret - 1);
+            if poll(&mut poll_fds, PollTimeout::NONE)? <= 0 {
+                error!("Poll error.");
+            }
+            if let Some(ev) = poll_fds[0].revents() {
+                if ev.contains(PollFlags::POLLIN) {
+                    let ret = tfd.read();
+                    if ret > 1 {
+                        error!("Missed {} interval(s)", ret - 1);
+                    }
+                    debug!("Time elapsed: {:?}", start.elapsed());
+                    current += time::Duration::from_secs(ret * self.init_params.interval);
+                    for (name, datatype) in self.collectors.iter_mut() {
+                        if datatype.is_static {
+                            continue;
+                        }
+
+                        datatype.collector_params.elapsed_time = start.elapsed().as_secs();
+
+                        aperf_collect_data.measure(
+                            name.clone() + "-collect",
+                            || -> Result<()> {
+                                datatype.collect_data()?;
+                                Ok(())
+                            },
+                        )?;
+                        aperf_collect_data.measure(name.clone() + "-print", || -> Result<()> {
+                            datatype.write_to_file()?;
+                            Ok(())
+                        })?;
+                    }
+                    let data_collection_time = time::Instant::now() - current;
+                    aperf_collect_data
+                        .data
+                        .insert("aperf".to_string(), data_collection_time.as_micros() as u64);
+                    debug!("Collection time: {:?}", data_collection_time);
+                    bincode::serialize_into(
+                        self.aperf_stats_handle.as_ref().unwrap(),
+                        &aperf_collect_data,
+                    )?;
+                }
             }
-            debug!("Time elapsed: {:?}", start.elapsed());
-            current += time::Duration::from_secs(ret * self.init_params.interval);
-            for (name, datatype) in self.collectors.iter_mut() {
-                if datatype.is_static {
-                    continue;
+            if let Some(ev) = poll_fds[1].revents() {
+                if ev.contains(PollFlags::POLLIN) {
+                    if let Ok(Some(siginfo)) = sfd.read_signal() {
+                        if siginfo.ssi_signo == signal::SIGINT as u32 {
+                            info!("Caught SIGINT. Exiting...");
+                        } else if siginfo.ssi_signo == signal::SIGTERM as u32 {
+                            info!("Caught SIGTERM. Exiting...");
+                        } else {
+                            error!("Caught unknown signal: {}. Exiting...", siginfo.ssi_signo);
+                        }
```

Comment:
We aren't registering an interest in any other signals, so this is impossible/dead code, I think it's fine to just omit it. Or replace it with a panic since it's not at all expected to happen.

Reply:
Yeah. We'll panic. Makes sense to become ugly if we're given a signal which we didn't register for.